contracts/misc/marketplaces/SeaportAdapter.sol

```diff
@@ -24,42 +24,41 @@ contract SeaportAdapter is IMarketplace {
 
     function getAskOrderInfo(bytes memory params)
         external
-        view
+        pure
         override
         returns (DataTypes.OrderInfo memory orderInfo)
     {
-        (
-            AdvancedOrder memory advancedOrder,
-            CriteriaResolver[] memory resolvers,
-            ,
-            address recipient
-        ) = abi.decode(
-                params,
-                (AdvancedOrder, CriteriaResolver[], bytes32, address)
-            );
+        (AdvancedOrder[] memory advancedOrders, , ) = abi.decode(
+            params,
+            (AdvancedOrder[], CriteriaResolver[], Fulfillment[])
+        );
         // support advanced order in the future
         require(
             // NOT criteria based and must be basic order
-            resolvers.length == 0 && isBasicOrder(advancedOrder),
+            advancedOrders.length == 2 &&
+                isBasicOrder(advancedOrders[0]) &&
+                isBasicOrder(advancedOrders[1]),
             Errors.INVALID_MARKETPLACE_ORDER
         );
-        // the person who listed NFT to sell
-        orderInfo.maker = advancedOrder.parameters.offerer;
-        require(
-            recipient == ADDRESSES_PROVIDER.getPool(),
-            Errors.INVALID_ORDER_TAKER
-        );
+        // the person who creates listing to sell NFT
+        orderInfo.maker = advancedOrders[0].parameters.offerer;
+        // the person who takes listing to buy NFT
+        orderInfo.taker = advancedOrders[1].parameters.offerer;
```

Comment:
should we remove the params.orderInfo.taker = msg.sender;  in executeBuyWithCredit  and executeBatchBuyWithCredit  ?